diff --git a/include/mbedtls/debug.h b/include/mbedtls/debug.h
index 29579964..9b8ebf50 100644
--- a/include/mbedtls/debug.h
+++ b/include/mbedtls/debug.h
@@ -40,28 +40,28 @@
 #define MBEDTLS_DEBUG_STRIP_PARENS( ... )   __VA_ARGS__
 
 #define MBEDTLS_SSL_DEBUG_MSG( level, args )                    \
-    mbedtls_debug_print_msg( ssl, level, __FILE__, __LINE__,    \
+    mbedtls_debug_print_msg( ssl, level, __FILENAME__, __LINE__,    \
                              MBEDTLS_DEBUG_STRIP_PARENS args )
 
 #define MBEDTLS_SSL_DEBUG_RET( level, text, ret )                \
-    mbedtls_debug_print_ret( ssl, level, __FILE__, __LINE__, text, ret )
+    mbedtls_debug_print_ret( ssl, level, __FILENAME__, __LINE__, text, ret )
 
 #define MBEDTLS_SSL_DEBUG_BUF( level, text, buf, len )           \
-    mbedtls_debug_print_buf( ssl, level, __FILE__, __LINE__, text, buf, len )
+    mbedtls_debug_print_buf( ssl, level, __FILENAME__, __LINE__, text, buf, len )
 
 #if defined(MBEDTLS_BIGNUM_C)
 #define MBEDTLS_SSL_DEBUG_MPI( level, text, X )                  \
-    mbedtls_debug_print_mpi( ssl, level, __FILE__, __LINE__, text, X )
+    mbedtls_debug_print_mpi( ssl, level, __FILENAME__, __LINE__, text, X )
 #endif
 
 #if defined(MBEDTLS_ECP_C)
 #define MBEDTLS_SSL_DEBUG_ECP( level, text, X )                  \
-    mbedtls_debug_print_ecp( ssl, level, __FILE__, __LINE__, text, X )
+    mbedtls_debug_print_ecp( ssl, level, __FILENAME__, __LINE__, text, X )
 #endif
 
 #if defined(MBEDTLS_X509_CRT_PARSE_C)
 #define MBEDTLS_SSL_DEBUG_CRT( level, text, crt )                \
-    mbedtls_debug_print_crt( ssl, level, __FILE__, __LINE__, text, crt )
+    mbedtls_debug_print_crt( ssl, level, __FILENAME__, __LINE__, text, crt )
 #endif
 
 #else /* MBEDTLS_DEBUG_C */
